app/models/spree/shipment_decorator.rb

```diff
@@ -1,7 +1,7 @@
 Spree::Shipment.class_eval do
   def self.exportable
-    query = joins(:order).where(spree_orders: { state: 'complete' })
-    query = query.where.not(spree_shipments: { state: 'pending' }) if Spree::Config.require_payment_to_ship
+    query = joins(:order).merge(Spree::Order.complete)
+    query = query.where.not(spree_shipments: { state: 'pending' }) if !Spree::Config.shipstation_capture_at_notification
```

Comment:
If we don't want `canceled` shipments then it needs to be global without regards of the capture at notification preference right ? so it should be then like this:

```ruby
    query = joins(:order).merge(Spree::Order.complete).where.not(spree_shipments: { state: 'canceled' })
    query = query.ready unless Spree::Config.shipstation_capture_at_notification
    query
```